pkg/router/routeapihelpers/validation_test.go

```diff
@@ -982,50 +1160,82 @@ func TestExtendedValidateRoute(t *testing.T) {
 					TLS: &routev1.TLSConfig{
 						Termination:   routev1.TLSTerminationEdge,
 						Certificate:   testCertificate,
-						Key:           testExpiredCertPrivateKey,
+						Key:           testCAUnknownCertificateKey,
 						CACertificate: testCACertificate,
 					},
 				},
 			},
 			expectedErrors: 1,
 		},
 		{
-			name: "Edge termination expired cert unknown CA",
+			name: "Edge termination cert with unknown CA",
 			route: &routev1.Route{
 				Spec: routev1.RouteSpec{
 					Host: "www.example.com",
 					TLS: &routev1.TLSConfig{
 						Termination: routev1.TLSTerminationEdge,
-						Certificate: testExpiredCAUnknownCertificate,
-						Key:         testExpiredCertPrivateKey,
+						Certificate: testCAUnknownCertificate,
+						Key:         testCAUnknownCertificateKey,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "Edge termination expired cert with unknown CA",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					Host: "think.different.test",
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						Certificate: testExpiredCert,
+						Key:         testExpiredCertKey,
 					},
 				},
 			},
 			expectedErrors: 0,
 		},
 		{
-			name: "Edge termination expired cert mismatched CA",
+			name: "Edge termination cert mismatched CA",
 			route: &routev1.Route{
 				Spec: routev1.RouteSpec{
 					Host: "www.example.com",
 					TLS: &routev1.TLSConfig{
 						Termination:   routev1.TLSTerminationEdge,
-						Certificate:   testExpiredCAUnknownCertificate,
-						Key:           testExpiredCertPrivateKey,
+						Certificate:   testCAUnknownCertificate,
+						Key:           testCAUnknownCertificateKey,
 						CACertificate: testCACertificate,
 					},
 				},
 			},
 			expectedErrors: 1,
 		},
 		{
-			name: "Edge termination expired cert key mismatch",
+			// Note: This odd test case is a symptom of the x509 cert Verify function returning
+			//       Expired err BEFORE returning mismatch err and the extended validation logic
+			//       ignores Expired err. So expectedErrors=0
+			name: "Edge termination with expired cert, ignoring CA mismatch",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					Host: "www.example.com",
+					TLS: &routev1.TLSConfig{
+						Termination:   routev1.TLSTerminationEdge,
+						Certificate:   testExpiredCert,
+						Key:           testExpiredCertKey,
+						CACertificate: testCACertificate,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "Edge termination cert key mismatch",
 			route: &routev1.Route{
 				Spec: routev1.RouteSpec{
 					Host: "www.example.com",
 					TLS: &routev1.TLSConfig{
 						Termination:   routev1.TLSTerminationEdge,
-						Certificate:   testExpiredCAUnknownCertificate,
+						Certificate:   testCAUnknownCertificate,
 						Key:           testPrivateKey,
 						CACertificate: testCACertificate,
```

Comment:
> > The test case expects only one error—perhaps for the same reason that "Edge termination with expired cert, ignoring CA mismatch" passes, in which case it might be helpful to add a comment saying as much.
> 
> Sorry, this is where I get confused. I thought you were taking about previously named `"Edge termination cert key mismatch"`, which returns 2 expectedErrors.

Sorry, I must have been mixing up test cases more misreading the diff.